exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/OtelToZipkinSpanTransformer.java

```diff
@@ -85,7 +86,8 @@ Span generateSpan(SpanData spanData) {
             .name(spanData.getName())
             .timestamp(toEpochMicros(spanData.getStartEpochNanos()))
             .duration(Math.max(1, endTimestamp - startTimestamp))
-            .localEndpoint(endpoint);
+            .localEndpoint(getLocalEndpoint(spanData))
+            .remoteEndpoint(getRemoteEndpoint(spanData));
```

Comment:
I was confused why `remoteEndpoint` on the resulting `Span` is set to null when `PEER_SERVICE` and the other attributes are missing. It turns out that `Span.Builder.remoteEndpoint(Endpoint)` ignores the call when the endpoint is empty.

Feels fragile to rely on this behavior. I think we should only attempt to resolve the remote endpoint when `SpanKind` is `CLIENT` or `PRODUCER`. Arguably, we should also only call the setter `remoteEndpoint` if we're able to resolve the endpoint based on the attribute.

Reply:
The idea here was providing whatever we have to Zipkin and let it decide if it wants it or not, i.e.: Zipkin can decide to drop the endpoint if `serviceName` is missing or it can use the IP instead of it if it wants to (it does not right now but maybe in a future version?).

But I see your point, let me add some logic around this.

Reply:
Fixed.